src/main/java/com/minecolonies/colony/Field.java

```diff
@@ -0,0 +1,467 @@
+
+    /**
+     * Tag to store the fields stage.
+     */
+    private static final String TAG_STAGE = "stage";
+
+    /**
+     * The max width/length of a field.
+     */
+    private static final int MAX_RANGE = 5;
+
+    /**
+     * The fields location.
+     */
+    private BlockPos location;
+
+    /**
+     * The colony of the field.
+     */
+    private final Colony colony;
+
+    /**
+     * Has the field be taken by any worker?
+     */
+    private boolean taken = false;
+
+    /**
+     * Checks if the field needsWork (Hoeig, Seedings, Farming etc)
+     */
+    private boolean needsWork = false;
+
+    /**
+     * Has the field been planted?
+     */
+    private FieldStage fieldStage = FieldStage.EMPTY;
+
+    /**
+     * The set seed type for the field.
+     */
+    private ItemSeeds seed;
+
+    /**
+     * The length to plus x of the field.
+     */
+    private int lengthPlusX;
+
+    /**
+     * The width to plus z of the seed.
+     */
+    private int widthPlusZ;
+
+    /**
+     * The length to minus xof the field.
+     */
+    private int lengthMinusX;
+
+    /**
+     * The width to minus z of the seed.
+     */
+    private int widthMinusZ;
+
+    /**
+     * The inventorySlot of the field.
+     */
+    private InventoryField inventory;
+
+    /**
+     * Describes the stage the field is in.
+     * Like if it has been hoed, planted or is empty.
+     */
+    public enum FieldStage
+    {
+        EMPTY,
+        HOED,
+        PLANTED
+    }
+
+
+     /**
+     * Private constructor to create field from NBT.
+     * @param colony the colony the field belongs to.
+     */
+    private Field(Colony colony)
+    {
+       this.colony = colony;
+    }
+
+    /**
+     * Creates an instance of our field container, this may be serve to open the GUI.
+     * @param inventory the field inventory.
+     * @param playerInventory the player inventory.
+     * @param world the world.
+     * @param location the position of the field.
+     */
+    public Field(InventoryField inventory, InventoryPlayer playerInventory, World world, BlockPos location)
+    {
+        this.colony = ColonyManager.getColony(world,location);
+        this.location = location;
+
+        this.inventory = inventory;
+
+        addSlotToContainer(new Slot(inventory, 0, 80, 34));
+
+        //Ddd player inventory slots
+        // Note: The slot numbers are within the player inventory and may be the same as the field inventory.
+        int i;
+        for (i = 0; i < 3; ++i)
+        {
+            for (int j = 0; j < 9; ++j)
+            {
+                addSlotToContainer(new Slot(playerInventory, j+i*9+9,
+                                            8+j*18, 84+i*18));
+            }
+        }
+
+        for (i = 0; i < 9; ++i)
+        {
+            addSlotToContainer(new Slot(playerInventory, i, 8 + i * 18,
+                                        142));
+        }
+        calculateSize(world,location.down());
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this call from a constructor to the overridable "calculateSize" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1699)